context/config_file.go

```diff
@@ -35,32 +31,126 @@ func parseConfigFile(fn string) (*configEntry, error) {
 }
 
 // ParseDefaultConfig reads the configuration file
-func ParseDefaultConfig() (*configEntry, error) {
+func ParseDefaultConfig() (*Config, error) {
 	return parseConfigFile(configFile())
 }
 
-func parseConfig(r io.Reader) (*configEntry, error) {
+type AuthConfig struct {
+	User  string
+	Token string
+}
+
+type HostConfig struct {
+	Host  string
+	Auths []*AuthConfig
+}
+
+type Config struct {
+	Root     *yaml.Node
+	Hosts    []*HostConfig
+	Editor   string
+	Protocol string
+}
+
+func (c *Config) ConfigForHost(hostname string) (*HostConfig, error) {
+	for _, hc := range c.Hosts {
+		if hc.Host == hostname {
+			return hc, nil
+		}
+	}
+	return nil, fmt.Errorf("could not find config entry for \"%s\"", hostname)
```

Comment:
Nit: you can easily display a quoted string value by using `%q` instead of `"%s"`